packages/kbn-cell-actions/src/actions/filter/filter_in.ts

```diff
@@ -69,18 +69,20 @@ export const addFilterIn = ({
   filterManager,
   fieldName,
   value,
+  negate,
   dataViewId,
 }: {
   filterManager: FilterManager | undefined;
   fieldName: string;
   value: DefaultActionsSupportedValue;
+  negate?: boolean;
   dataViewId?: string;
 }) => {
   if (filterManager != null) {
     const filter = createFilter({
       key: fieldName,
       value,
-      negate: isEmptyFilterValue(value),
+      negate: negate ?? isEmptyFilterValue(value),
```

Comment:
This change is not ideal here. 
I understand why it is needed, but the `addFilterIn` and `addFilterOut` functions are helpers that abstract the `negate` logic for each type of filter. If we need to force the `negate` value to them there's no point in using them to begin with, passing `negate` both _filterIn_ and _filterOut_ helpers will do exactly the same thing, so there's no reason to distinguish between them.

I think what we should do to solve this is extract the logic in between and call it directly from the Lens filter action:

1 - Extract an `addFilter` function in a common file (e.g. _'./add_filter'_) with:
```
interface AddFilterParams {
  filterManager: FilterManager;
  key: string;
  value: DefaultActionsSupportedValue;
  negate: boolean;
  dataViewId?: string;
}

export const addFilter = ({ filterManager, key, value, negate, dataViewId }: AddFilterParams) => {
  filterManager.addFilters(createFilter({ key, value, negate, dataViewId }));
};
```
2 - Update `addFilterIn` and `addFilterOut` to call this generic `addFilter` with their own `negate` flag.
 
3 - In the same _add_filter_ file, create an `addExistsFilter` function, that will not receive any `value`:
```
interface AddExistsFilterParams {
  filterManager: FilterManager;
  key: string;
  negate: boolean;
  dataViewId?: string;
}
export const addExistsFilter = ({ filterManager, key, negate, dataViewId }: AddExistsFilterParams) => {
  filterManager.addFilters(createExistsFilter({ key, negate, dataViewId }));
};
```

4 - Finally, the Lens filter action will call `addFilter` and `addExistsFilter` functions directly.

Does it make sense?

Reply:
Yes, thanks for the tips! But I haven't update `x-pack/plugins/security_solution/public/actions/filter/lens/create_action.ts` to call `addFilter`, as that'd change the existing behaviour for other types of cell value.

Reply:
Yes, let's keep using `addFilterIn` and `addFilterOut` in the "non-counter" case, even better, thanks for doing this!